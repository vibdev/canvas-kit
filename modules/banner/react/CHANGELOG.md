# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.25](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.24...@workday/canvas-kit-react-banner@0.1.25) (2019-06-20)


### Bug Fixes

* broad stroke accessibility fixes ([9803412](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/commits/9803412))





## [0.1.24](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.23...@workday/canvas-kit-react-banner@0.1.24) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.23](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.22...@workday/canvas-kit-react-banner@0.1.23) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.22](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.21...@workday/canvas-kit-react-banner@0.1.22) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.21](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.20...@workday/canvas-kit-react-banner@0.1.21) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.20](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.19...@workday/canvas-kit-react-banner@0.1.20) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.19](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.18...@workday/canvas-kit-react-banner@0.1.19) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.18](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.17...@workday/canvas-kit-react-banner@0.1.18) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.16...@workday/canvas-kit-react-banner@0.1.17) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.15...@workday/canvas-kit-react-banner@0.1.16) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.14...@workday/canvas-kit-react-banner@0.1.15) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.13...@workday/canvas-kit-react-banner@0.1.14) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.12...@workday/canvas-kit-react-banner@0.1.13) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.11...@workday/canvas-kit-react-banner@0.1.12) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.10...@workday/canvas-kit-react-banner@0.1.11) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.9...@workday/canvas-kit-react-banner@0.1.10) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.8...@workday/canvas-kit-react-banner@0.1.9) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.7...@workday/canvas-kit-react-banner@0.1.8) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.6...@workday/canvas-kit-react-banner@0.1.7) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.5...@workday/canvas-kit-react-banner@0.1.6) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.4...@workday/canvas-kit-react-banner@0.1.5) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.3...@workday/canvas-kit-react-banner@0.1.4) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.2...@workday/canvas-kit-react-banner@0.1.3) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.1...@workday/canvas-kit-react-banner@0.1.2) (2019-05-06)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## 0.1.1 (2019-04-30)

**Note:** Version bump only for package @workday/canvas-kit-react-banner
