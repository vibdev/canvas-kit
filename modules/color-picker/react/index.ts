import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
