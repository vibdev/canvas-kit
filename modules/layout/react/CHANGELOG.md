# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.16...@workday/canvas-kit-react-layout@0.1.17) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.15...@workday/canvas-kit-react-layout@0.1.16) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.14...@workday/canvas-kit-react-layout@0.1.15) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.13...@workday/canvas-kit-react-layout@0.1.14) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.12...@workday/canvas-kit-react-layout@0.1.13) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.11...@workday/canvas-kit-react-layout@0.1.12) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.10...@workday/canvas-kit-react-layout@0.1.11) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.9...@workday/canvas-kit-react-layout@0.1.10) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.8...@workday/canvas-kit-react-layout@0.1.9) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.7...@workday/canvas-kit-react-layout@0.1.8) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.6...@workday/canvas-kit-react-layout@0.1.7) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.5...@workday/canvas-kit-react-layout@0.1.6) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.4...@workday/canvas-kit-react-layout@0.1.5) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.3...@workday/canvas-kit-react-layout@0.1.4) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.2...@workday/canvas-kit-react-layout@0.1.3) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.1...@workday/canvas-kit-react-layout@0.1.2) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.0...@workday/canvas-kit-react-layout@0.1.1) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





# 0.1.0 (2019-04-10)


### Features

* **layout:** added fixed width prop for columns ([fe4a860](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/fe4a860))
* **layouts:** added canvas spacing to layouts ([607ed59](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/607ed59))
* **layouts:** added spacing feature ([f388d0d](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/f388d0d))
* **layouts:** content -> capWidth ([93efa87](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/93efa87))
* **layouts:** fixed type errors ([463447a](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/463447a))
* **layouts:** initial commit for layouts component ([ae29eae](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/ae29eae))
* **layouts:** limited columns to 1 through 12 ([3cbacfb](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/3cbacfb))
* **layouts:** made column children optional ([26831f3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/26831f3))
* **layouts:** removed padding from first and last element on column ([9a3e9d3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/9a3e9d3))
* **layouts:** updated readme ([e10218d](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/e10218d))
* **layouts:** updated readme ([8fdc4ce](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/8fdc4ce))
