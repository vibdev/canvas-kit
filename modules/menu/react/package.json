{
  "name": "@workday/canvas-kit-react-menu",
  "version": "0.0.24",
  "description": "A container for navigation or action items",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "menu"
  ],
  "devDependencies": {
    "@material-ui/core": "^3.9.2",
    "@types/react-transition-group": "^2.0.14",
    "@workday/canvas-system-icons-web": "^1.0.1",
    "react-transition-group": "^2.5.0"
  },
  "dependencies": {
    "@types/uuid": "^3.4.4",
    "@workday/canvas-kit-react-card": "^0.3.23",
    "@workday/canvas-kit-react-common": "^0.7.5",
    "@workday/canvas-kit-react-core": "^1.10.6",
    "@workday/canvas-kit-react-icon": "^1.7.7",
    "@workday/design-assets-types": "^0.2.3",
    "emotion": "^9.2.12",
    "react-emotion": "^9.2.12",
    "uuid": "^3.3.2"
  },
  "peerDependencies": {
    "react": ">= 16.3 < 17"
  }
}
