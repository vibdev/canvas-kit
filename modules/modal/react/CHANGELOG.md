# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/modal/compare/@workday/canvas-kit-react-modal@0.0.1...@workday/canvas-kit-react-modal@0.0.2) (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal





## 0.0.1 (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal
