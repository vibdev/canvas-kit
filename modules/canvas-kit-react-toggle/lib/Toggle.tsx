import * as React from 'react';
import styled from 'react-emotion';
import {focusRing} from '@workday/canvas-kit-react-common';
import {colors, depth} from '@workday/canvas-kit-react-core';

export interface ToggleProps {
  checked: boolean;
  disabled?: boolean;
  onChange?: (e: React.SyntheticEvent) => void;
  value?: string;
  inputRef?: React.Ref<HTMLInputElement>;
  inputProps?: {};
}

const circleSize = 12;
const toggleWidth = 32;
const toggleHeight = 16;
const translateLength = toggleWidth - toggleHeight;

const ToggleInput = styled('input')({
  height: toggleHeight,
  width: toggleWidth,
  opacity: 0,
  position: 'absolute',
  margin: 0,
  borderRadius: 999,
});

const ToggleBackground = styled('div')<ToggleProps>(({checked, disabled}) => ({
  width: toggleWidth,
  height: toggleHeight,
  backgroundColor: disabled ? colors.soap200 : checked ? colors.blueberry500 : colors.soap400,
  borderRadius: 999,
  display: 'flex',
  alignItems: 'center',
  transition: 'background-color 0.25s linear',
  padding: '0px 2px',
  boxSizing: 'border-box',
}));
const ToggleCircle = styled('div')<ToggleProps>(({checked}) => ({
  width: circleSize,
  height: circleSize,
  borderRadius: 100,
  boxShadow: depth[1].boxShadow,
  backgroundColor: colors.frenchVanilla100,
  transform: checked ? `translateX(${translateLength}px)` : 'translateX(0)',
  transition: 'transform 0.25s linear',
}));

const ToggleContainer = styled('div')<ToggleProps>(({disabled}) => ({
  display: 'inline-flex',
  alignItems: 'center',
  cursor: disabled ? 'no-allowed' : 'pointer',
  borderRadius: 999,
  '&:focus-within': {
    '&:not([disabled])': {
      outline: 'none',
      ...focusRing(2, 2),
    },
  },
}));

export default class ToggleSwitch extends React.Component<ToggleProps> {
  public static defaultProps = {
    checked: false,
  };
  public render() {
    const {onChange, checked, disabled, value, inputRef, ...inputProps} = this.props;
    return (
      <ToggleContainer {...this.props}>
        <ToggleInput
          innerRef={inputRef}
          value={value}
          disabled={disabled}
          tabIndex={0}
          checked={checked}
          onChange={onChange}
          type="checkbox"
          role="checkbox"
          {...inputProps}
        />
        <ToggleBackground checked={checked}>
          <ToggleCircle checked={checked} />
        </ToggleBackground>
      </ToggleContainer>
    );
  }
}
