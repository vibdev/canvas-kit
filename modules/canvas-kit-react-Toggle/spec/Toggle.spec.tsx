import * as React from 'react';
import Toggle from '../lib/Toggle';
import * as renderer from 'react-test-renderer';

describe('Toggle Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(
      <Toggle
        isChecked={false}
        onClick={() => {
          /* foo */
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders toggle on', () => {
    const component = renderer.create(
      <Toggle
        isChecked={true}
        onClick={() => {
          /* foo */
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders toggle with different background color', () => {
    const component = renderer.create(
      <Toggle
        isChecked={true}
        onClick={() => {
          /* foo */
        }}
        toggleBackgroundColor="#000"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
