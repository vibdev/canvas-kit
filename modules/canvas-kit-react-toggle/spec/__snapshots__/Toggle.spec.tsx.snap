// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshots renders as expected 1`] = `
.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 999px;
}

.emotion-3:focus-within:not([disabled]) {
  outline: none;
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-0 {
  height: 16px;
  width: 32px;
  opacity: 0;
  position: absolute;
  margin: 0;
  border-radius: 999px;
}

.emotion-2 {
  width: 32px;
  height: 16px;
  background-color: #dfe2e6;
  border-radius: 999px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.25s linear;
  transition: background-color 0.25s linear;
  padding: 0px 2px;
  box-sizing: border-box;
}

.emotion-1 {
  width: 12px;
  height: 12px;
  border-radius: 100px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 0.25s linear;
  -webkit-transition: transform 0.25s linear;
  transition: transform 0.25s linear;
}

<div
  checked={false}
  className="emotion-3"
  onChange={[Function]}
>
  <input
    checked={false}
    className="emotion-0"
    onChange={[Function]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={false}
    className="emotion-2"
  >
    <div
      checked={false}
      className="emotion-1"
    />
  </div>
</div>
`;

exports[`Toggle Snapshots renders toggle disabled checked 1`] = `
.emotion-0 {
  height: 16px;
  width: 32px;
  opacity: 0;
  position: absolute;
  margin: 0;
  border-radius: 999px;
}

.emotion-2 {
  width: 32px;
  height: 16px;
  background-color: #005cb9;
  border-radius: 999px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.25s linear;
  transition: background-color 0.25s linear;
  padding: 0px 2px;
  box-sizing: border-box;
}

.emotion-1 {
  width: 12px;
  height: 12px;
  border-radius: 100px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 0.25s linear;
  -webkit-transition: transform 0.25s linear;
  transition: transform 0.25s linear;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: no-allowed;
  border-radius: 999px;
}

.emotion-3:focus-within:not([disabled]) {
  outline: none;
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

<div
  checked={true}
  className="emotion-3"
  disabled={true}
  onChange={[Function]}
>
  <input
    checked={true}
    className="emotion-0"
    disabled={true}
    onChange={[Function]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={true}
    className="emotion-2"
  >
    <div
      checked={true}
      className="emotion-1"
    />
  </div>
</div>
`;

exports[`Toggle Snapshots renders toggle disabled unchecked 1`] = `
.emotion-0 {
  height: 16px;
  width: 32px;
  opacity: 0;
  position: absolute;
  margin: 0;
  border-radius: 999px;
}

.emotion-2 {
  width: 32px;
  height: 16px;
  background-color: #dfe2e6;
  border-radius: 999px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.25s linear;
  transition: background-color 0.25s linear;
  padding: 0px 2px;
  box-sizing: border-box;
}

.emotion-1 {
  width: 12px;
  height: 12px;
  border-radius: 100px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 0.25s linear;
  -webkit-transition: transform 0.25s linear;
  transition: transform 0.25s linear;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: no-allowed;
  border-radius: 999px;
}

.emotion-3:focus-within:not([disabled]) {
  outline: none;
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

<div
  checked={false}
  className="emotion-3"
  disabled={true}
  onChange={[Function]}
>
  <input
    checked={false}
    className="emotion-0"
    disabled={true}
    onChange={[Function]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={false}
    className="emotion-2"
  >
    <div
      checked={false}
      className="emotion-1"
    />
  </div>
</div>
`;

exports[`Toggle Snapshots renders toggle on 1`] = `
.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 999px;
}

.emotion-3:focus-within:not([disabled]) {
  outline: none;
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-0 {
  height: 16px;
  width: 32px;
  opacity: 0;
  position: absolute;
  margin: 0;
  border-radius: 999px;
}

.emotion-2 {
  width: 32px;
  height: 16px;
  background-color: #005cb9;
  border-radius: 999px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.25s linear;
  transition: background-color 0.25s linear;
  padding: 0px 2px;
  box-sizing: border-box;
}

.emotion-1 {
  width: 12px;
  height: 12px;
  border-radius: 100px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 0.25s linear;
  -webkit-transition: transform 0.25s linear;
  transition: transform 0.25s linear;
}

<div
  checked={true}
  className="emotion-3"
  onChange={[Function]}
>
  <input
    checked={true}
    className="emotion-0"
    onChange={[Function]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={true}
    className="emotion-2"
  >
    <div
      checked={true}
      className="emotion-1"
    />
  </div>
</div>
`;

exports[`Toggle Snapshots renders toggle with name and value 1`] = `
.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 999px;
}

.emotion-3:focus-within:not([disabled]) {
  outline: none;
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-0 {
  height: 16px;
  width: 32px;
  opacity: 0;
  position: absolute;
  margin: 0;
  border-radius: 999px;
}

.emotion-2 {
  width: 32px;
  height: 16px;
  background-color: #005cb9;
  border-radius: 999px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.25s linear;
  transition: background-color 0.25s linear;
  padding: 0px 2px;
  box-sizing: border-box;
}

.emotion-1 {
  width: 12px;
  height: 12px;
  border-radius: 100px;
  box-shadow: 0px 2px 4px 0 rgba(0,0,0,0.08);
  background-color: #ffffff;
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
  -webkit-transition: -webkit-transform 0.25s linear;
  -webkit-transition: transform 0.25s linear;
  transition: transform 0.25s linear;
}

<div
  checked={true}
  className="emotion-3"
  disabled={false}
  onChange={[Function]}
  value="user"
>
  <input
    checked={true}
    className="emotion-0"
    disabled={false}
    onChange={[Function]}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
    value="user"
  />
  <div
    checked={true}
    className="emotion-2"
  >
    <div
      checked={true}
      className="emotion-1"
    />
  </div>
</div>
`;
